shopelectro/models.py

```diff
@@ -172,15 +162,23 @@ def payment_type_name(self):
     def set_positions(self, cart):
         """Save cart's state into Order instance."""
         self.save()
+        self.revenue = 0
         for id_, position in cart:
+            print(position['price'], position['purchase_price'])
+            print(position['quantity'])
+            import sys
+            sys.stdout.flush()
+            price_diff = position['price'] - position['purchase_price']
+            self.revenue += price_diff * position['quantity']
```

Comment:
several thoughts about `revenue` field:
- it's not clear to calc revenue field in `set_positions` method
- it's not clear to keep revenue in DB. Because every order will have a state when this revenue was either set or not
- Cart is more good place to cal revenue: it focused only on positions and it's prices. Order = cart + credentials

So, i suggest to create Cart's `revenue` or `total_revenue` property: it already has total_price and total_quantity props

Reply:
>it's not clear to calc revenue field in set_positions method

From  `set_positions` docs: `Save cart's state into Order instance.`. May be it is not the best name for this method, but it falls into the semantic bounds

>it's not clear to keep revenue in DB

We have to calculate a revenue from a cart and save it to db, because when we process a successful order, we will clear the cart, although we still need this revenue in next steps

>Cart is more good place to cal revenue

We can calculate a revenue in the cart, but we still have to save it in an order.

I will move a calculation of revenue to `Cart.revenue`

Reply:
@artemiy312 

> We would put a revenue to a session

in this case `Cart.clear` method will clear the whole session instead of one revenue field.
It can be confusing state between real order sessions.

-----

As of other points: you are right here. Seems set_positions is the best place to save revenue in our arch.
But semantically it's still not the best place in the world because of semantical binding between Cart.revenue and Cart.set_positions().
It's the problem in our arch:
- now we suggest that Cart.positions == CartSession (contains all it's data). It's not true
- we should create separated CartSession and models.Cart classes
- they should have methods CartSession.as_model and models.Cart.as_session
- after `cart_session.clear()` call a binded `cart_model` should preserve it's state

Create pdd task with the link on this comment plz. We can continue discussion in it's comments if it's needed